spacepy/plot/utils.py

```diff
@@ -1400,3 +1401,149 @@ def show_used(fig=None):
+    :method:`~matplotlib.axes.Axes.annotate`.  Arrows will be spread evenly
+    over the line using the number of points in the line as the metric for
+    spacing.  For example, if a line has 120 points and 3 arrows are requested,
+    an arrow will be added at point number 30, 60, and 90.
+    Arrow color and alpha is obtained from the parent line.
+
+    Parameters
+    ==========
+    lines : :class:`~matplotlib.lines.Line2D`, a list/tuple, or :class:`~matplotlib.collections.LineCollection`
+        A single line or group of lines on which to place the arrows.
+        Arrows inherent color and transparency (alpha) from the line on which
+        they are placed.
+
+
+    Other Parameters
+    ================
+    n : integer
+        Number of arrows to add to each line; defaults to 3.
+
+    size : integer
+        The size of the arrows in points.  Defaults to 12.
+
+    style : string
+        Set the style of the arrow, e.g. '->' (default) or '-|>'
+
+    dorestrict : boolean
+        If True (default), only points along the line within the current
+        limits of the axes will be considered when distributing arrows.
+
+    positions : Nx2 array
+        N must be the number of lines provided via the argument *lines*.
+        If provided, only one arrow will be placed per line.
+        *positions* sets the explicit location of each arrow for each line as
+        X-Y space in Axes coordinates.
+
+    Returns
+    =======
+    None
+
+    Notes
+    =====
+    The algorithm works by dividing the line in to *n*+1 segements and 
+    placing an arrow between each segement, endpoints excluded.  Arrows span
+    the shortest distance possible, i.e., two adjacent points along a line.
+    For lines that are long spatially but sparse in points, the arrows will
+    have long tails that may extend beyond axes bounds.  For explicit positions,
+    the arrow is placed at the point on the curve closest to that position
+    and the exact position is not always attainable.  A maximum number of arrows
+    equal to one-half of the number of points in a line per line will be
+    created, so not all lines will receive *n* arrows.
+
+    Example
+    =======
+    >>> import matplotlib.pyplot as plt
+    >>> import numpy as np
+    >>> x = np.arange(1, 10, .01)
+    >>> y = np.sin(x)
+    >>> line = plt.plot(x,y)
+    >>> add_arrows(line, n=15, style='-|>')
+
+    '''
+    from numpy import floor, argmin
+    from matplotlib.collections import LineCollection
+    from matplotlib.lines import Line2D
+    
+    # Convert our line, lines, or LineCollection into
+    # a series of arrays with the x/y data.
+    # Also, grab the axes, colors, and line alpha.
+    if isinstance(lines, Line2D):
+        # Single line: turn into a list for iteration.
+        ax   = lines.axes
+        data = [lines.get_xydata()]
+        cols = [lines.get_color()]
+        alph = [lines.get_alpha()]
+    elif isinstance(lines, (tuple, list)):
```

Comment:
So this assumes that any tuple or list passed in has line objects, so if you pass in the wrong list you get an error that's maybe not helpful to most users.

E.g., 
```~/projects/env/lib/python3.6/site-packages/spacepy/plot/utils.py in add_arrows(lines, n, size, style, dorestr
ict, positions)
   1489     elif isinstance(lines, (tuple, list)):
   1490         # Multiple lines: convert into list of xy points.
-> 1491         ax   = lines[0].axes
   1492         data = [x.get_xydata() for x in lines]
   1493         cols = [x.get_color()  for x in lines]

AttributeError: 'int' object has no attribute 'axes'
```

Can this whole if/elif/else block be done with try/except so that it's more general but captures the main behaviours being assumed by checking for the specific types?

Reply:
I'm being dense, but could you be more specific here about employing the try-except blocks here?  Or nesting try-except blocks inside the `elif`s to catch errors? 

Best way may be to simply add try-except inside the `isinstance(lines, (tuple, list))` portion to ensure each entry is a Line2d obect...

Reply:
Per offline discussion, this is addressed now.  I think we're in good shape.